packages/api-contract/src/Abi/index.ts

```diff
@@ -103,13 +111,24 @@ export class Abi {
       const typeSpec = spec.returnType.unwrapOr(null);
 
       return this.#createMessage(spec, index, {
+        isDefault: spec.default.isTrue,
         isMutating: spec.mutates.isTrue,
         isPayable: spec.payable.isTrue,
         returnType: typeSpec
           ? this.registry.lookup.getTypeDef(typeSpec.type)
           : null
       });
     });
+
+    for (const [key, value] of this.metadata.spec.environment.entries()) {
+      const typeSpec = value.toPrimitive();
+
+      if (typeof typeSpec === 'object' && typeSpec !== null && 'type' in typeSpec) {
+        this.environment.set(key, this.registry.lookup.getTypeDef(typeSpec.type as number));
+      } else {
+        this.environment.set(key, typeSpec as number);
+      }
+    }
```

Comment:
The issue here is that because of the conversions, some types may be completely invalid since they point to defaults.

The suggested approach (and here is hoping it never changes from JSON), is to mark the types as `Option<...>` for instance, `accountId: 'Option<ContractTypeSpec>',`. That way when converting from old -> new, it will be `isNone`.

Reply:
created `Option<ContractEnvironment>` for the entire thing, now returns an empty map for older versions.

Reply:
Thank you.